### PR TITLE
Enhancment of scsi_add_remove test script

### PR DESCRIPTION
**Pull request by Naresh-ibm**
existing test script cannot be run on vfc disk which does not has pci_address.
and also, script doesn't has verification part after remove or added the scsi
disk. So modified the script to run using wwid or pci domains and also added
the verification part to it.

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
[scsi_add_remove_job_log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/3445588/scsi_add_remove_job_log.txt)

File: io/disk/scsi_add_remove.py.data/README.txt [hunk 1]
```diff
@@ -1,7 +1,8 @@
 This Test removes and adds back a scsi device in all the specified PCI domains
-specified in the 'multiplexer' file.
+or wwids specified in the 'multiplexer' file.
```
Comment (narasimhan-v) [inline io/disk/scsi_add_remove.py.data/README.txt]: @Naresh-ibm also mention here:
if wwids are provided, pci devices are ignored.

File: io/disk/scsi_add_remove.py [hunk 2]
```diff
@@ -33,44 +37,90 @@ def setUp(self):
...
+            self.log.info("\nDeleting %s = %s\n" % (device, scsi_id))
+            genio.write_file("/sys/block/%s/device/delete" % device, "1")
+            time.sleep(5)
+            if self.is_exists_scsi_device(device) is True:
+                self.err_paths.append(device)
+            else:
+                self.log.info("\n%s = %s deleted\n" % (device, scsi_id))
+            self.log.info("\nadding back %s = %s\n" % (device, scsi_id))
+            part = "scsi add-single-device %s" % scsi_id
+            genio.write_file("/proc/scsi/scsi", part)
```
Comment (narasimhan-v) [inline io/disk/scsi_add_remove.py]: Is this same as `process.run("echo scsi add-single-device %s > /proc/scsi/scsi", scsi_id)` ?

Comment (narasimhan-v) [reply io/disk/scsi_add_remove.py]: Just asking for confirmation.

File: io/disk/scsi_add_remove.py [hunk 3]
```diff
@@ -33,44 +37,90 @@ def setUp(self):
...
+        self.device_list = []
         smm = SoftwareManager()
         if not smm.check_installed("lsscsi") and not smm.install("lsscsi"):
             self.cancel("lsscsi is not installed")
-        self.pci_device = self.params.get("pci_device", default=None)
-        if not self.pci_device:
-            self.cancel("Please provide PCI address for which you \
-                        want to run the test")
+        self.wwids = self.params.get('wwids', default='')
+        self.pci_device = self.params.get("pci_device", default='').split(',')
```
Comment (narasimhan-v) [inline io/disk/scsi_add_remove.py]: The split should happen after checking for presence, similar to self.wwids

File: io/disk/scsi_add_remove.py (hunk 2 above)
Comment (Naresh-ibm) [reply io/disk/scsi_add_remove.py]: yes

File: io/disk/scsi_add_remove.py (hunk 3 above)
Comment (Naresh-ibm) [reply io/disk/scsi_add_remove.py]: it works fine even if we have the split at this place. 
So I think it should be fine.

Comment (narasimhan-v) [reply io/disk/scsi_add_remove.py]: @Naresh-ibm 
```
>>> bool(''.split(','))
True
>>> bool('')
False
>>>
```

**Comment by Naresh-ibm** [comment]
closing this in favour of #1249

**Pull request closed without merging**